frontend/src/components/settings/route/RouteSetting.js

```diff
@@ -0,0 +1,193 @@
+import React, {useEffect, useState} from "react";
+import FormControl from "@material-ui/core/FormControl";
+import FormLabel from "@material-ui/core/FormLabel";
+import FormGroup from "@material-ui/core/FormGroup";
+import FormHelperText from "@material-ui/core/FormHelperText";
+import Grid from "@material-ui/core/Grid";
+import AddIcon from "@material-ui/icons/Add";
+import ClearIcon from "@material-ui/icons/Clear";
+import Button from "@material-ui/core/Button";
+import {green, grey, red} from "@material-ui/core/colors";
+
+import {makeStyles} from "@material-ui/core/styles";
+import {ThemeProvider} from "styled-components";
+import { createMuiTheme } from "@material-ui/core/styles";
+
+import RoutePoint from "./RoutePoint";
+import RouteHints from "./RouteHints";
+import {routeServices} from "./routeServices";
+
+const theme = createMuiTheme();
+
+const useStyles = makeStyles((theme) => ({
+    formContainer: {
+        marginTop: '18px',
+    },
+    addButton: {
+        display: 'float',
+        float: 'right',
+    },
+}))
+
+export default function RouteSetting(props) {
+    const classes = useStyles()
+
+    const [route, setRoute] = useState(props.route)
+    const [errors, setErrors] = useState({
+        start: false,
+        end: false,
+        points: []
+    });
+
+    // initial array hook with input name values for points (not start and end)
+    const pointsValues = []
+    for (let i in props.route.points) {
+        pointsValues[i] = "point" + i
+    }
+    const [points, setPoints] = useState(pointsValues)
+
+    // check required values for start and end only, other points are optional
+    const { start, end } = route;
+    const error = [ start, end ].filter((v) => v).length === 0;
+
+    // set values of start and end to empty string and delete other points
+    const handleReset = event => {
+        setRoute({start: "", end: "", points: []})
+        setErrors({start: false, end: false, points: []})
+        setPoints([])
+    }
+
+    /* useEffect(() => {
+        console.log("R:", route, "E:", errors, "P", points)
+    }, [route, errors, points]) */
+
+    // update hooks for route, errors and points: index of end ist
```

Comment:
Refactoring: use function handleChange(event) for start and end and handleChangePoint(event, index) 

const handleChange = (event) => {
        const value = event.target.value
        const errorValue = (value === "" || !routeServices.checkRoutePointValue(value))
        setRoute({...route, [item]: value});
        setErrors({...errors, [item]: errorValue});
}
const handleChangePoint = (event, index) => {
        const value = event.target.value
        const errorValue = (value === "" || !routeServices.checkRoutePointValue(value))

        const newRoutePoints = route.points
        newRoutePoints[index] = value
        setRoute({...route, [item]: newRoutePoints});

        const newErrors = errors.points
        newErrors[index] = errorValue;
        setErrors({...errors, [item]: newErrors});
    }